File: app/src/main/java/com/example/androidcodelabapp/model/GithubUsers.java
package com.example.androidcodelabapp.model;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.SerializedName;

public class GithubUsers implements Parcelable {
    public static final Creator<GithubUsers> CREATOR = new Creator<GithubUsers>() {
        @Override
        public GithubUsers createFromParcel(Parcel source) {
            return new GithubUsers(source);
        }

        @Override
        public GithubUsers[] newArray(int size) {
            return new GithubUsers[size];
        }
    };

    @SerializedName("avatar_url")
    String profileImage;

    @SerializedName("login")
    String userName;

    @SerializedName("html_url")
    String profile;

    @SerializedName("company")
    String organization;

    protected GithubUsers(Parcel source) {
        profileImage = source.readString();
        userName = source.readString();
        organization = source.readString();
        profile = source.readString();
    }


    public String getProfileImage() {
        return profileImage;
    }

    public String getUserName() {
        return userName;
    }

    public String getOrganization() {
        return organization;
    }

    public String getProfile() {
        return profile;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(profileImage);
        dest.writeString(userName);
        dest.writeString(organization);
        dest.writeString(profile);
    }

}

File: app/src/main/java/com/example/androidcodelabapp/model/GithubUsers.kt
package com.example.androidcodelabapp.model

import android.os.Parcel
import android.os.Parcelable

import com.google.gson.annotations.SerializedName

class GithubUsers( profileImage: String?, userName: String?, profile: String?,
                   organization: String?) : Parcelable {
    @SerializedName("avatar_url")
    var profileImage: String? = null

    @SerializedName("login")
    var userName: String? = null

    @SerializedName("html_url")
    var profile: String? = null

    @SerializedName("company")
    var organization: String? = null

    init {
        this.profileImage = profileImage
        this.userName = userName
        this.profile = profile
        this.organization = organization
    }

    constructor(parcel: Parcel) : this(parcel.readString(), parcel.readString(),
            parcel.readString(), parcel.readString())

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(profileImage)
        parcel.writeString(userName)
        parcel.writeString(profile)
        parcel.writeString(organization)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<GithubUsers> {
        override fun createFromParcel(parcel: Parcel): GithubUsers {
            return GithubUsers(parcel)
        }

        override fun newArray(size: Int): Array<GithubUsers?> {
            return arrayOfNulls(size)
        }
    }
}
